Support Domain mode of WildFly and EAP servers

mgencur opened the pull request:
The version for WildFly 10 is 1.1.0.Beta1 on purpose. There was a problem with starting the server with version 1.0.0 and it was fixed in 1.1.0.Beta1.

<!-- Reviewable:start -->
---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/arquillian/arquillian-container-chameleon/44)

<!-- Reviewable:end -->

aslakknutsen commented on src/main/java/org/arquillian/container/chameleon/spi/model/Target.java:
```diff
@@ -5,7 +5,7 @@
 
     public static enum Type {
-        Remote, Managed, Embedded, Default;
+        Remote, Managed, Embedded, Domain, Default;
```
hmm.. This would hard code domain to be Managed(or whatever Type: domain is configured to). Domain can technically be both Remote and Managed.

Wondering if this would be better as a 'server', e.g. WildFly Domain:10.0.0.Final:managed

mgencur replied:
Right. I was thinking about adding Domain_Managed, Domain_Remote, etc. but it didn't look very good. Specifying "domain" in the server part sounds better. Let me look at this.

mgencur commented:
Hey @aslakknutsen , fixed ! I think it looks better now.

bartoszmajsak commented:
Thanks! Would it make sense to add a profile for managed containers in domain mode? WDYT?

mgencur commented:
Hi, it is already there in the PR. The usage is following: WildFly Domain:10.0.0.Final:managed

bartoszmajsak commented:
Must have missed that. So we are all good for merging from your point of view?

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/arquillian/arquillian-container-chameleon/44#-:-KM8XXa07BSldeUyr07k:264747564)*
<!-- Sent from Reviewable.io -->

mgencur commented:
Yes, this is ready from my POV. It's been here for a few months.

bartoszmajsak commented:
I think my intention in the previous comment was to add those new containers to `all` profile for the Maven build.

I've tried to run following
```
mvn clean install -Darq.container.chameleon.configuration.chameleonTarget="WildFly Domain:10.0.0.Final:managed"
```

against latest master (you will need to resolve simple merge conflict in `containers.yaml` if you want to run it smoothly), but this leads to test failures.

@mgencur can you verify if that is also a problem on your side?

bartoszmajsak commented:
I would like to release Chameleon soon. @mgencur can you check on your side if there is an issue  which I encountered?

I would like to release new version soon :)

MatousJobanek commented:
Hi, the problem could be caused by the version of container adapter - the correct version should be 2.0.0.Final. IIRC there was a rename of class in core for embedded containers. This means that container adapter 1.1.x embedded won't work with EAP 7 and WF10.

MatousJobanek commented:
Anyway, if the version of the container adapter is fixed, the test is still failing with this exception:
`DeploymentException: Could not determine the server-group to deploy the archive to.`
This can be solved by specifying `@TargetsContainer("main-server-group")` for the deployment there in SimpleDeploymentTestCase

mgencur commented:
Hi, I've rebased my branch on latest master and changed the adapter version to 2.0.0.Final. Thanks for the tips Matous! 
I've also added another test that runs for domain mode and added this run to the "all" profile. Let me know if there's anything else I should do. Cheers, Martin

MatousJobanek commented on pom.xml:
```diff
@@ -321,15 +322,15 @@
                 </configuration>
               </execution>
               <execution>
-                <id>payara41-managed</id>
```
May I know why you replaced (removed) the original payara41-managed profile?

bartoszmajsak replied:
Maybe a merge issue?

MatousJobanek commented:
Landed in https://github.com/arquillian/arquillian-container-chameleon/commit/296dc1609fc54eaa4947b2f9dd1c82ced4452b8d :rocket: Thanks a lot for your contribution :+1:
